src/ripple/app/reporting/ReportingETL.cpp

```diff
@@ -904,21 +901,20 @@ ReportingETL::ReportingETL(Application& app, Stoppable& parent)
         // if not passed via command line, check config for start sequence
         if (!startSequence_)
         {
-            std::pair<std::string, bool> start = section.find("start_sequence");
-            if (start.second)
+            auto const optStartSeq = section.get("start_sequence");
+            if (optStartSeq)
             {
-                startSequence_ = std::stoi(start.first);
+                startSequence_ = std::stoi(*optStartSeq);
```

Comment:
If the `start_sequence` value supplied in the configuration is invalid, `std::stoi` will cause the program to crash without providing helpful output as to the cause. The same applies for `flushInterval_` and `numMarkers_`.

Reply:
@undertome, thanks for pointing that out.  @HowardHinnant, do you see any downside to replacing all uses of  `std::stoi()` in the code base with calls to `std::from_chars()`?  Or can you suggest a better replacement?

Reply:
`from_chars` sounds good to me as long as we always check the return for error, and if we have something sane to do on error.  If we don't have a sane error recovery strategy, then letting `stoi` throw doesn't sound unreasonable.

Reply:
@undertome, with @HowardHinnant's guidance in mind I looked harder at your question.  Everything gets more difficult when you take (consistent) error handling into account.

If we look higher in `ReportingETL`'s constructor we see that `throw`ing without logging is the expected error handling mechanism.  That makes a certain amount of sense.  The log file may not have been set up this early while assembling the server.

As you point out, `std::stoi()` does throw, but the error that it throws does not provide enough context for a user to figure out what's going on.  

I came up with an approach that may improve the error handling in `ReportingETL`'s constructor.  You can see it here:

https://github.com/scottschurr/rippled/commit/5a36c6b271aa12ad87edbf188a133ab509728224

Unfortunately there is zero unit test coverage in that portion of the code.  So I lack confidence in the proposed change.  If @cjcobb23 looks at it and thinks it's okay I'm good with including the commit.  Otherwise I think the change is too risky.

Reply:
This patch looks good @scottschurr . Crashing with a useful error message makes sense.

Reply:
Thanks @cjcobb23, I've added that patch to this branch.